### DISCUSSION TITLE
Detecting if app.py invoked via Flask Migrate

### DISCUSSION THREAD

**Discussion by diloreto** [miguelgrinberg/Flask-Migrate#453] [Q&A]
Hi All,

Hopefully a simple question - I want to be able to run the Migrate function only when the app.py is being invoked by flask migrate. If we're running the app to start the server, no need to run any migrations. Any ideas/direction on how to do this?

For example:

```
# invoked by flask migrate means user called flask migrate -m "message" on the CLI
if( script_invoked_by_flask_migrate ):
    migrate = Migrate(app, db)
else:
    start_server()
```

Any help is appreciated!

**Comment by miguelgrinberg** [comment]
There's a chicken and egg problem. The `flask migrate` command is only registered when you create the `Migrate` instance.

**Reply by diloreto** [reply]
Thanks @miguelgrinberg - is there a canonical way to include the `Migrate` instantiation only for migrate needs (and not normal server deployment)? I tried inspecting the stack:

`from inspect import stack`

and using properties/functions of stack, but as you can imagine python sees no difference between:

`flask run`

and 

`flask migrate -m "message"`

Any ideas?

**Reply by miguelgrinberg** [reply]
Why don't you use Alembic directly if you don't want to pollute your `flask` command with stuff?

**Reply by diloreto** [reply]
We actually ended up separating the `app.py` into `app.py` and `migrate.py`, and removed the **Flask-Migrate** code in `app.py`, and setting an env var on virtualenv starutp. Thank you!
